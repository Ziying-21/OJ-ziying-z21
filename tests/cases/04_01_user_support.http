###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":1}
###
# Got response:
# HTTP 404
# content-length: 65
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# access-control-allow-credentials: true
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:18 GMT
# 
# {"code":3,"message":"User 1 not found.","reason":"ERR_NOT_FOUND"}
###
# Send request:
GET http://127.0.0.1:12345/users HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 24
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# access-control-allow-credentials: true
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:18 GMT
# 
# [{"id":0,"name":"root"}]
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"name":"user1"}
###
# Got response:
# HTTP 200
# content-length: 23
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# access-control-allow-credentials: true
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:18 GMT
# 
# {"id":1,"name":"user1"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":1}
###
# Got response:
# HTTP 200
# content-length: 377
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# access-control-allow-credentials: true
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:18 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Waiting","time":0},{"id":1,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:19:19.243Z","id":0,"result":"Waiting","score":0.0,"state":"Queueing","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":1},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/0 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 426
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# access-control-allow-credentials: true
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:19 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":123448},{"id":1,"info":"","memory":0,"result":"Accepted","time":66453}],"created_time":"2022-09-09T02:19:19.243Z","id":0,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":0,"source_code":"fn main() { println!(\"Hello World!\"); }","user_id":1},"updated_time":"2022-09-09T02:19:19.437Z"}
###
# Send request:
GET http://127.0.0.1:12345/users HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 48
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# access-control-allow-credentials: true
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:19 GMT
# 
# [{"id":0,"name":"root"},{"id":1,"name":"user1"}]
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"id":1,"name":"new_user1"}
###
# Got response:
# HTTP 200
# content-length: 27
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# access-control-allow-credentials: true
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:20 GMT
# 
# {"id":1,"name":"new_user1"}
###
# Send request:
GET http://127.0.0.1:12345/users HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 52
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# access-control-allow-credentials: true
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:20 GMT
# 
# [{"id":0,"name":"root"},{"id":1,"name":"new_user1"}]
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"name":"new_user1"}
###
# Got response:
# HTTP 400
# content-length: 91
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# access-control-allow-credentials: true
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:20 GMT
# 
# {"code":1,"message":"User name 'new_user1' already exists","reason":"ERR_INVALID_ARGUMENT"}
###
# Send request:
POST http://127.0.0.1:12345/users HTTP/1.1
content-type: application/json

{"id":2,"name":"user2"}
###
# Got response:
# HTTP 404
# content-length: 65
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# content-type: application/json
# access-control-allow-credentials: true
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:20 GMT
# 
# {"code":3,"message":"User 2 not found.","reason":"ERR_NOT_FOUND"}
