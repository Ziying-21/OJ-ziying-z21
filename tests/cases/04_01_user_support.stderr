[2022-09-09T02:19:18Z INFO  actix_server::builder] Starting 8 workers
[2022-09-09T02:19:18Z INFO  actix_server::server] Actix runtime found; starting in Actix runtime
[2022-09-09T02:19:19Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET / HTTP/1.1" 404 0 "-" "-" 0.000176
[2022-09-09T02:19:19Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 404 65 "-" "-" 0.000186
[2022-09-09T02:19:19Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /users HTTP/1.1" 200 24 "-" "-" 0.000097
[2022-09-09T02:19:19Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 200 23 "-" "-" 0.000383
[2022-09-09T02:19:19Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 377 "-" "-" 0.000399
[2022-09-09T02:19:20Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/0 HTTP/1.1" 200 426 "-" "-" 0.000323
[2022-09-09T02:19:20Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /users HTTP/1.1" 200 48 "-" "-" 0.000085
[2022-09-09T02:19:20Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 200 27 "-" "-" 0.000877
[2022-09-09T02:19:20Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /users HTTP/1.1" 200 52 "-" "-" 0.000072
[2022-09-09T02:19:20Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 400 91 "-" "-" 0.000096
[2022-09-09T02:19:20Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /users HTTP/1.1" 404 65 "-" "-" 0.000090
